sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerListDetails.java

```diff
@@ -42,13 +45,53 @@ public BlobContainerListDetails setRetrieveMetadata(boolean retrieveMetadata) {
         return this;
     }
 
+    /**
+     * Whether deleted containers should be returned.
+     *
+     * @return a flag indicating whether deleted containers should be returned
+     */
+    public boolean getRetrieveDeleted() {
+        return this.retrieveDeleted;
+    }
+
+    /**
+     * Whether deleted containers should be returned.
+     *
+     * @param retrieveDeleted Flag indicating whether deleted containers should be returned.
+     * @return the updated ContainerListDetails object
+     */
+    public BlobContainerListDetails setRetrieveDeleted(boolean retrieveDeleted) {
+        this.retrieveDeleted = retrieveDeleted;
+        return this;
+    }
+
     /**
      * @return the listing flags
+     * @deprecated use {@link #toIncludeTypes()}
      */
+    @Deprecated
     public ListBlobContainersIncludeType toIncludeType() {
         if (this.retrieveMetadata) {
             return ListBlobContainersIncludeType.METADATA;
         }
         return null;
     }
+
+    /**
+     * @return the listing flags
```

Comment:
There needs to be much more JavaDoc here.

Reply:
I've added more.

Reply:
I ended up converting this to private method inside client.